index.js

```diff
@@ -99,10 +102,27 @@ export class Placement {
       this.target.appendChild(element);
 
       return this;
+    }).then((placement) => {
+      // Detect when the ad is in the viewport
+      // Add the view pixel to the DOM to count the view
+
+      var viewport_detection = setInterval((element) => {
+        // Verge can be off by 1-2 pixels
+        // A fudge factor of ~3 is needed for the case where the ad
+        // is hidden off the side of the screen by a sliding sidebar
+        // For example, if the right side of the ad is at x=0
+        // or the left side of the ad is at the right side of the viewport
+        if (placement.response && placement.response.view_url && verge.inViewport(element, -3)) {
+          // This ad was seen!
+          var pixel = document.createElement("img");
+          pixel.src = placement.response.view_url;
+          pixel.className = "ea-pixel";
+          element.appendChild(pixel);
+
+          clearInterval(viewport_detection);
+        }
+      }, 100, placement.target);
```

Comment:
I assume this is 100ms? Seems reasonable, but we could go even lower if we wanted probably if we're worried about it.

Reply:
It is 100ms. I don't know if we really need to go lower. I believe the IAB guidelines are that an ad should be visible for a full second although I can't seem to find that guideline right now. One possibility is to both trigger it immediately and every 100ms.

Reply:
I did end up finding it although there's conflicting guidelines.

Firstly there's this ([source](https://www.iab.com/wp-content/uploads/2015/06/MRC-Viewable-Ad-Impression-Measurement-Guideline.pdf)):

> Requirements for Viewable Display Advertising Impressions
In addition to the above requirements, Viewable Display Ad Impressions are
counted when the following criteria are met:
> • Pixel Requirement: Greater than or equal to 50% of the pixels in the
advertisement were on an in-focus browser tab on the viewable space of
the browser page, and
> • Time Requirement: The time the pixel requirement is met was greater
than or equal to one continuous second, post ad render.

Secondly there's this ([source](https://www.iab.com/wp-content/uploads/2017/11/Desktop-Display-Impression-Measurement-Guidelines-US-MMTF-Final-v1.1.pdf)):

> An Ad Impression across all display marketing channels is the measurement of responses
from an ad delivery system to an ad request from the user's browser, which is filtered for
invalid traffic and is recorded at a point as late as possible in the process of delivery of the
creative material to the user's browser. The ad must be loaded and at minimum begin to
render in order to count it as a valid ad impression. 

The latter of these is more recent and seems more correctly applicable.

Reply:
Yea, it's better than what we have now either way :)